From b2a541f9701816c394cd8dcbf2b242b6e5274afe Mon Sep 17 00:00:00 2001
From: roqqit <roqqit@proton.me>
Date: Wed, 20 Nov 2024 13:54:33 -0800
Subject: [PATCH] [DogeCash] Disable avalanche until it is ready (#57)

Avalanche per-se is not an issue, however, if it's enabled, it disables deep reorgs. This means that by default, nodes would get stuck on a shorter chain on a deep reorg.

The simplest way to fix this is to disable Avalanche by default until we're positive Dogecoin will activate it.

Co-authored-by: Tobias Ruck <ruck.tobias@gmail.com>

diff --git a/src/avalanche/avalanche.h b/src/avalanche/avalanche.h
index 3c661117d2019672644b183d7b9083272ba2dcd8..b3ddb83246ff5d411711436c7ef41cc9439b04ec 100644
--- a/src/avalanche/avalanche.h
+++ b/src/avalanche/avalanche.h
@@ -12,7 +12,7 @@
 /**
  * Is avalanche enabled by default.
  */
-static constexpr bool AVALANCHE_DEFAULT_ENABLED = true;
+static constexpr bool AVALANCHE_DEFAULT_ENABLED = false;
 
 /**
  * Conflicting proofs cooldown time default value in seconds.
diff --git a/test/functional/abc_feature_avalanche_options.py b/test/functional/abc_feature_avalanche_options.py
index 4de6a625f27ca88fc0c37582b68c4b46366810b1..aa887dfd5d738bba3cbbda1edc7fcd60164b7a2c 100644
--- a/test/functional/abc_feature_avalanche_options.py
+++ b/test/functional/abc_feature_avalanche_options.py
@@ -14,7 +14,11 @@ class AvalancheOptionsTest(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
-        self.extra_args = [[]]
+        self.extra_args = [
+            [
+                "-avalanche=1",
+            ]
+        ]
 
     def run_test(self):
         node = self.nodes[0]
@@ -29,6 +33,7 @@ class AvalancheOptionsTest(BitcoinTestFramework):
             self.restart_node(
                 0,
                 extra_args=[
+                    "-avalanche=1",
                     "-maxconnections=10",
                     "-maxavalancheoutbound=20",
                 ],
@@ -50,6 +55,7 @@ class AvalancheOptionsTest(BitcoinTestFramework):
 
         node.assert_start_raises_init_error(
             extra_args=[
+                "-avalanche=1",
                 f"-avaproof={proof.serialize().hex()}",
                 f"-avamasterkey={bytes_to_wif(privkey.get_bytes())}",
                 f"-conf={os.path.join(node.datadir, 'bitcoin_nobind.conf')}",
@@ -61,6 +67,7 @@ class AvalancheOptionsTest(BitcoinTestFramework):
 
         node.assert_start_raises_init_error(
             extra_args=[
+                "-avalanche=1",
                 f"-avaproof={proof.serialize().hex()}",
                 f"-avamasterkey={bytes_to_wif(privkey.get_bytes())}",
                 "-proxy=127.0.01:9050",
@@ -70,6 +77,7 @@ class AvalancheOptionsTest(BitcoinTestFramework):
 
         node.assert_start_raises_init_error(
             extra_args=[
+                "-avalanche=1",
                 f"-avaproof={proof.serialize().hex()}",
                 f"-avamasterkey={bytes_to_wif(privkey.get_bytes())}",
                 "-i2psam=127.0.01:7656",
@@ -79,6 +87,7 @@ class AvalancheOptionsTest(BitcoinTestFramework):
 
         node.assert_start_raises_init_error(
             extra_args=[
+                "-avalanche=1",
                 f"-avaproof={proof.serialize().hex()}",
                 f"-avamasterkey={bytes_to_wif(privkey.get_bytes())}",
                 "-onlynet=ipv6",
diff --git a/test/functional/abc_feature_minerfund.py b/test/functional/abc_feature_minerfund.py
index 8d1faf53efab04a69d3d7395659f0421c405b564..ceb4d4bb8bdcd4f100fce6665a8bc4338df954ff 100644
--- a/test/functional/abc_feature_minerfund.py
+++ b/test/functional/abc_feature_minerfund.py
@@ -52,6 +52,7 @@ class MinerFundTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-enableminerfund",
             ],
             [],
@@ -150,7 +151,7 @@ class MinerFundTest(BitcoinTestFramework):
 
     def test_without_avalanche(self):
         self.log.info("Test the behavior when avalanche is completely disabled")
-        self.restart_node(0, extra_args=["-enableminerfund"])
+        self.restart_node(0, extra_args=["-avalanche=1", "-enableminerfund"])
         self.restart_node(1, extra_args=["-avalanche=0"])
         self.connect_nodes(0, 1)
 
diff --git a/test/functional/abc_feature_persist_avapeers.py b/test/functional/abc_feature_persist_avapeers.py
index 728bef3f956355605c4e66be22a501744734b66d..a286b978cd6481871fd5140fc9a86958dd19fb18 100644
--- a/test/functional/abc_feature_persist_avapeers.py
+++ b/test/functional/abc_feature_persist_avapeers.py
@@ -25,6 +25,7 @@ class AvalanchePersistAvapeers(BitcoinTestFramework):
         self.num_nodes = 1
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/abc_feature_proof_cleanup.py b/test/functional/abc_feature_proof_cleanup.py
index 34b029078e31cd1d2bf848d2c722df16f56f085f..515dc6a9bcf1ea50d52128a92dbe81ed7bc34eac 100644
--- a/test/functional/abc_feature_proof_cleanup.py
+++ b/test/functional/abc_feature_proof_cleanup.py
@@ -31,6 +31,7 @@ class ProofsCleanupTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
             ]
diff --git a/test/functional/abc_mining_rtt.py b/test/functional/abc_mining_rtt.py
index 14eb3850fa43d2d32efec0233353e094b88f008c..6c9c0616be0e0d3af6850340dbb86e1ff01bea40 100644
--- a/test/functional/abc_mining_rtt.py
+++ b/test/functional/abc_mining_rtt.py
@@ -46,6 +46,7 @@ class AbcMiningHeartbeatTest(BitcoinTestFramework):
         self.extra_args = [
             [
                 "-enablertt",
+                "-avalanche",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/abc_mining_stakingrewards.py b/test/functional/abc_mining_stakingrewards.py
index 37e161cd62be9a0b0c1ca9293e25b61f2be74409..65c84cd72cc97fc6cc7aea2ee2af5c35c78e4b20 100644
--- a/test/functional/abc_mining_stakingrewards.py
+++ b/test/functional/abc_mining_stakingrewards.py
@@ -48,6 +48,7 @@ class AbcMiningStakingRewardsTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/abc_p2p_avalanche_cooldown.py b/test/functional/abc_p2p_avalanche_cooldown.py
index 4144a0ae8a68d37b66d8c76b2c112932059f5bda..0a2fcc40fccd0af58ce298fdb1416c5bd051c78b 100644
--- a/test/functional/abc_p2p_avalanche_cooldown.py
+++ b/test/functional/abc_p2p_avalanche_cooldown.py
@@ -30,6 +30,7 @@ class AvalancheCooldownTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-enableminerfund",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
diff --git a/test/functional/abc_p2p_avalanche_peer_discovery.py b/test/functional/abc_p2p_avalanche_peer_discovery.py
index b9e516ce4a26494ee80437ee102cbfa514c1334f..be71c1f339e093d29c3b6a3968ab9cd43662beeb 100644
--- a/test/functional/abc_p2p_avalanche_peer_discovery.py
+++ b/test/functional/abc_p2p_avalanche_peer_discovery.py
@@ -54,6 +54,7 @@ class AvalanchePeerDiscoveryTest(BitcoinTestFramework):
         self.num_nodes = 1
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=3",
                 "-persistavapeers=0",
diff --git a/test/functional/abc_p2p_avalanche_policy_minerfund.py b/test/functional/abc_p2p_avalanche_policy_minerfund.py
index daf832c650017ed65948afe8b3debf9408bb1869..a667050fe87875636ee94a67e8a68757d4c080fc 100644
--- a/test/functional/abc_p2p_avalanche_policy_minerfund.py
+++ b/test/functional/abc_p2p_avalanche_policy_minerfund.py
@@ -40,6 +40,7 @@ class AvalancheMinerFundTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-enableminerfund",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
diff --git a/test/functional/abc_p2p_avalanche_policy_rtt.py b/test/functional/abc_p2p_avalanche_policy_rtt.py
index b89e40de534c87e93d101b4a7ae7bf1490e5b699..34d57a53e2743733894806fe80260f4adb22a7c7 100644
--- a/test/functional/abc_p2p_avalanche_policy_rtt.py
+++ b/test/functional/abc_p2p_avalanche_policy_rtt.py
@@ -28,6 +28,7 @@ class AvalancheRTTTest(BitcoinTestFramework):
         self.extra_args = [
             [
                 "-enablertt",
+                "-avalanche",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
@@ -35,7 +36,9 @@ class AvalancheRTTTest(BitcoinTestFramework):
                 "-avaminavaproofsnodecount=0",
                 "-persistavapeers=0",
             ],
-            [],
+            [
+                "-avalanche"
+            ],
         ]
 
     def check_rtt_policy(self):
diff --git a/test/functional/abc_p2p_avalanche_policy_stakingrewards.py b/test/functional/abc_p2p_avalanche_policy_stakingrewards.py
index 90f59a2cc4ebe91fc53d096fb76bfb91a71bc61b..e86cf7ad17dd65731e3d8e20c2e337efe472f82f 100644
--- a/test/functional/abc_p2p_avalanche_policy_stakingrewards.py
+++ b/test/functional/abc_p2p_avalanche_policy_stakingrewards.py
@@ -38,6 +38,7 @@ class ABCStakingRewardsPolicyTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/abc_p2p_avalanche_proof_voting.py b/test/functional/abc_p2p_avalanche_proof_voting.py
index c281db528e9ec5269283c7507bf227411acea486..cc1593dfbb544281d320a238dd61c0fde0c2ec58 100644
--- a/test/functional/abc_p2p_avalanche_proof_voting.py
+++ b/test/functional/abc_p2p_avalanche_proof_voting.py
@@ -45,6 +45,7 @@ class AvalancheProofVotingTest(BitcoinTestFramework):
         self.peer_replacement_cooldown = 2000
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 f"-avaproofstakeutxoconfirmations={self.avaproof_stake_utxo_confirmations}",
                 f"-avalancheconflictingproofcooldown={self.conflicting_proof_cooldown}",
@@ -346,6 +347,7 @@ class AvalancheProofVotingTest(BitcoinTestFramework):
         self.restart_node(
             0,
             extra_args=[
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 f"-avaproofstakeutxoconfirmations={self.avaproof_stake_utxo_confirmations}",
                 "-avacooldown=0",
@@ -556,6 +558,7 @@ class AvalancheProofVotingTest(BitcoinTestFramework):
         self.restart_node(
             0,
             extra_args=[
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 f"-avaproofstakeutxoconfirmations={self.avaproof_stake_utxo_confirmations}",
                 "-avalancheconflictingproofcooldown=0",
diff --git a/test/functional/abc_p2p_avalanche_quorum.py b/test/functional/abc_p2p_avalanche_quorum.py
index f91331b6fe7f5314b140298ac6c3e5fcf3ae55c9..fc0c3ad74c3dfe375c44afcd1d735bff9ec29f02 100644
--- a/test/functional/abc_p2p_avalanche_quorum.py
+++ b/test/functional/abc_p2p_avalanche_quorum.py
@@ -30,6 +30,7 @@ class AvalancheQuorumTest(BitcoinTestFramework):
         self.min_avaproofs_node_count = 8
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/abc_p2p_avalanche_remote_proofs.py b/test/functional/abc_p2p_avalanche_remote_proofs.py
index b25bb7c9d016acc479b94ab9c42a1a717f6a3dc9..661db9bbf6b7ed2e7ca59ba74e480cf5f573c1a7 100644
--- a/test/functional/abc_p2p_avalanche_remote_proofs.py
+++ b/test/functional/abc_p2p_avalanche_remote_proofs.py
@@ -36,6 +36,7 @@ class AvalancheRemoteProofsTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=10000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avaminquorumstake=1000000",
diff --git a/test/functional/abc_p2p_avalanche_transaction_voting.py b/test/functional/abc_p2p_avalanche_transaction_voting.py
index 4d6dfed4cebf483b1677d5cfc731cc56565cc597..cf3c3b6d411ad14f3b1be51e0614ce8549a6b5ed 100644
--- a/test/functional/abc_p2p_avalanche_transaction_voting.py
+++ b/test/functional/abc_p2p_avalanche_transaction_voting.py
@@ -48,6 +48,8 @@ class AvalancheTransactionVotingTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
+                "-avalanchepreconsensus=1",
                 "-avacooldown=0",
                 "-avaproofstakeutxoconfirmations=1",
                 # Low enough for coinbase transactions to be staked in valid proofs
diff --git a/test/functional/abc_p2p_avalanche_voting.py b/test/functional/abc_p2p_avalanche_voting.py
index 6648644f57dd52dcdcde10c9e6686a72e3cf98a3..ab327bc24094aefb6eb263a5342291f53038977e 100644
--- a/test/functional/abc_p2p_avalanche_voting.py
+++ b/test/functional/abc_p2p_avalanche_voting.py
@@ -35,6 +35,7 @@ class AvalancheTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
@@ -44,6 +45,7 @@ class AvalancheTest(BitcoinTestFramework):
                 "-avalanchestakingpreconsensus=0",
             ],
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/abc_p2p_compactproofs.py b/test/functional/abc_p2p_compactproofs.py
index 28bb052c1860613c6313805f313c5683b193e2e6..56891c9153767d80b450f8243e57bdf93876768a 100644
--- a/test/functional/abc_p2p_compactproofs.py
+++ b/test/functional/abc_p2p_compactproofs.py
@@ -56,6 +56,7 @@ class CompactProofsTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/abc_p2p_getavaaddr.py b/test/functional/abc_p2p_getavaaddr.py
index 85a8b40311c31282cf607e047698e090c067842d..32e626701b39d971b0b0c9b4863c42982c350652 100644
--- a/test/functional/abc_p2p_getavaaddr.py
+++ b/test/functional/abc_p2p_getavaaddr.py
@@ -107,6 +107,7 @@ class AvaAddrTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/abc_p2p_proof_inventory.py b/test/functional/abc_p2p_proof_inventory.py
index 7f0e9e5b007255887d351ed0dd637f3201d8bcb6..494db1674c48116ef207c2f4f5b2ea4149550657 100644
--- a/test/functional/abc_p2p_proof_inventory.py
+++ b/test/functional/abc_p2p_proof_inventory.py
@@ -58,6 +58,7 @@ class ProofInventoryTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=2",
                 "-avacooldown=0",
@@ -156,7 +157,9 @@ class ProofInventoryTest(BitcoinTestFramework):
             ],
         )
 
-        self.restart_node(0, ["-avaproofstakeutxodustthreshold=1000000"])
+        self.restart_node(
+            0, ["-avalanche=1", "-avaproofstakeutxodustthreshold=1000000"]
+        )
 
         peer = node.add_p2p_connection(P2PInterface())
         msg = msg_avaproof()
@@ -203,7 +206,8 @@ class ProofInventoryTest(BitcoinTestFramework):
 
         chainwork = int(self.nodes[-1].getblockchaininfo()["chainwork"], 16)
         restart_nodes_with_proof(
-            self.nodes[-1:], extra_args=[f"-minimumchainwork={chainwork + 100:#x}"]
+            self.nodes[-1:],
+            extra_args=["-avalanche=1", f"-minimumchainwork={chainwork + 100:#x}"],
         )
 
         # Add an inbound so the node proof can be registered and advertised
diff --git a/test/functional/abc_rpc_addavalanchenode.py b/test/functional/abc_rpc_addavalanchenode.py
index 93395071ef1718cdab9f90e8a18b564466beaa31..9b2b22cd8c3fcd42ec26a231aa3433bdf1b7bf7d 100644
--- a/test/functional/abc_rpc_addavalanchenode.py
+++ b/test/functional/abc_rpc_addavalanchenode.py
@@ -32,6 +32,7 @@ class AddAvalancheNodeTest(BitcoinTestFramework):
         self.num_nodes = 1
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/abc_rpc_avalancheproof.py b/test/functional/abc_rpc_avalancheproof.py
index e3f2ff5ba7a6f7ff8048946e18e7daf810415ee9..e3069defb3113bc239e5d1dafefeec18255b9081 100644
--- a/test/functional/abc_rpc_avalancheproof.py
+++ b/test/functional/abc_rpc_avalancheproof.py
@@ -61,6 +61,7 @@ class AvalancheProofTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 f"-avaproofstakeutxodustthreshold={PROOF_DUST_THRESHOLD}",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avalancheconflictingproofcooldown=0",
diff --git a/test/functional/abc_rpc_buildavalancheproof.py b/test/functional/abc_rpc_buildavalancheproof.py
index fc38b40428e856dd043e13694d8cb5eb9a9d8eb7..f1959a04d5a7616a0f6699bc085f9e912db6112e 100644
--- a/test/functional/abc_rpc_buildavalancheproof.py
+++ b/test/functional/abc_rpc_buildavalancheproof.py
@@ -14,7 +14,9 @@ from test_framework.wallet_util import bytes_to_wif
 class BuildAvalancheProofTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
-        self.extra_args = [["-avaproofstakeutxoconfirmations=1", "-avacooldown=0"]]
+        self.extra_args = [
+            ["-avalanche=1", "-avaproofstakeutxoconfirmations=1", "-avacooldown=0"]
+        ]
 
     def run_test(self):
         node = self.nodes[0]
diff --git a/test/functional/abc_rpc_getavalancheinfo.py b/test/functional/abc_rpc_getavalancheinfo.py
index b4677217f5f44f561119d674d2828ccadd604ad8..21b1053e690bc8c7b6575b2c81926467fb06a606 100644
--- a/test/functional/abc_rpc_getavalancheinfo.py
+++ b/test/functional/abc_rpc_getavalancheinfo.py
@@ -45,6 +45,7 @@ class GetAvalancheInfoTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 f"-avalancheconflictingproofcooldown={self.conflicting_proof_cooldown}",
                 "-avacooldown=0",
                 "-avaminquorumstake=250000000",
diff --git a/test/functional/abc_rpc_getavalanchepeerinfo.py b/test/functional/abc_rpc_getavalanchepeerinfo.py
index 969398b92d6f8e9a491bbdbc435af9ba924e112d..3298ec3e76e5269bf919b53951e626d8e5eaae91 100644
--- a/test/functional/abc_rpc_getavalanchepeerinfo.py
+++ b/test/functional/abc_rpc_getavalanchepeerinfo.py
@@ -26,6 +26,7 @@ class GetAvalanchePeerInfoTest(BitcoinTestFramework):
         self.num_nodes = 1
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/abc_rpc_getavalancheproofs.py b/test/functional/abc_rpc_getavalancheproofs.py
index 406b4487436e4357916cbefbfcc7078597db185f..771da1ef3110967efdfbc71056afd61eb6db7737 100644
--- a/test/functional/abc_rpc_getavalancheproofs.py
+++ b/test/functional/abc_rpc_getavalancheproofs.py
@@ -26,6 +26,7 @@ class GetAvalancheProofsTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 f"-avalancheconflictingproofcooldown={self.conflicting_proof_cooldown}",
                 "-avaproofstakeutxoconfirmations=2",
                 "-avacooldown=0",
diff --git a/test/functional/abc_rpc_invalidateavalancheproof.py b/test/functional/abc_rpc_invalidateavalancheproof.py
index 241288f8d2bd0fe78f73f5917322c5f1bc519777..af15a2bed0428b807299be5a3acfb22ff689c502 100644
--- a/test/functional/abc_rpc_invalidateavalancheproof.py
+++ b/test/functional/abc_rpc_invalidateavalancheproof.py
@@ -29,6 +29,7 @@ class InvalidateAvalancheProofTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avalancheconflictingproofcooldown=0",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/abc_rpc_isfinal.py b/test/functional/abc_rpc_isfinal.py
index 36be780cc7da90bc46c778eeb7f439fd99e2b887..0eefebe0691cc9c18c148bab49dfd52b50e59846 100644
--- a/test/functional/abc_rpc_isfinal.py
+++ b/test/functional/abc_rpc_isfinal.py
@@ -27,6 +27,7 @@ class AvalancheIsFinalTest(BitcoinTestFramework):
         self.num_nodes = 1
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/chronik_avalanche.py b/test/functional/chronik_avalanche.py
index 8c6f1094718c01ef7171910f9dcbd1a4f85e6de3..54cd9653fed0985b46c690d2c70a9f5b0d0ca54f 100644
--- a/test/functional/chronik_avalanche.py
+++ b/test/functional/chronik_avalanche.py
@@ -33,6 +33,7 @@ class ChronikAvalancheTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/chronik_tx_removal_order.py b/test/functional/chronik_tx_removal_order.py
index 42596a7abcd5f961d63963fdcde57861c4de13c5..306b99125323acfafe7f8a6cbd95870fdef92942 100644
--- a/test/functional/chronik_tx_removal_order.py
+++ b/test/functional/chronik_tx_removal_order.py
@@ -32,7 +32,7 @@ class ChronikTxRemovalOrder(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
-        self.extra_args = [["-chronik"]]
+        self.extra_args = [["-chronik", "-avalanche"]]
         self.rpc_timeout = 240
 
     def skip_test_if_missing_module(self):
diff --git a/test/functional/chronik_ws.py b/test/functional/chronik_ws.py
index 01ccc1f19c5e0b76f11c1d1846bdf6a4f35aca1b..a11edc59c226ee26aef7013290344bf1162d98cc 100644
--- a/test/functional/chronik_ws.py
+++ b/test/functional/chronik_ws.py
@@ -46,6 +46,7 @@ class ChronikWsTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/chronik_ws_ordering.py b/test/functional/chronik_ws_ordering.py
index 0602394945bef849e82dcdc8c8e1ce83555fd3dc..e1b7513e387b132beadad0ea897b669145b7a194 100644
--- a/test/functional/chronik_ws_ordering.py
+++ b/test/functional/chronik_ws_ordering.py
@@ -30,6 +30,7 @@ class ChronikWsOrdering(BitcoinTestFramework):
         self.extra_args = [
             [
                 "-chronik",
+                "-avalanche",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/chronik_ws_script.py b/test/functional/chronik_ws_script.py
index 5dc665737562166e1c5d784d632285607f6b048b..5b7ade489f04c98bfc1d4e09acedcbe98338a836 100644
--- a/test/functional/chronik_ws_script.py
+++ b/test/functional/chronik_ws_script.py
@@ -36,6 +36,7 @@ class ChronikWsScriptTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [
+                "-avalanche",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/p2p_add_connections.py b/test/functional/p2p_add_connections.py
index 2fac166c55e9d4570dda2da7e7702bdaa3fb07e0..9cf760bcd5ed96faa8e3ecbfdf2def5cc27f9dcb 100644
--- a/test/functional/p2p_add_connections.py
+++ b/test/functional/p2p_add_connections.py
@@ -33,6 +33,7 @@ class P2PAddConnections(BitcoinTestFramework):
         self.num_nodes = 2
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxoconfirmations=1",
                 f"-maxavalancheoutbound={MAX_AVALANCHE_OUTBOUND_CONNECTIONS}",
             ],
diff --git a/test/functional/p2p_eviction.py b/test/functional/p2p_eviction.py
index b1a15098cf8ce8d1a616fd82423ec35613e69344..ac224ffb49d790ae14225261fae0470d8a89899f 100644
--- a/test/functional/p2p_eviction.py
+++ b/test/functional/p2p_eviction.py
@@ -66,6 +66,7 @@ class P2PEvict(BitcoinTestFramework):
         # avalanche availability score
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-maxconnections=188",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
diff --git a/test/functional/p2p_inv_download.py b/test/functional/p2p_inv_download.py
index 8258eca8277db906c009d7f5a6a32ecf6a3ee604..ba95423c7e9a73e05ec93ad4d32ce18b954af258 100644
--- a/test/functional/p2p_inv_download.py
+++ b/test/functional/p2p_inv_download.py
@@ -132,6 +132,7 @@ class InventoryDownloadTest(BitcoinTestFramework):
         self.num_nodes = 2
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-avacooldown=0",
diff --git a/test/functional/rpc_net.py b/test/functional/rpc_net.py
index b48994fd37793861753fcf76ed5d426fb86ef18e..6c5b0738c7e887339d653892d3560682c590dd59 100644
--- a/test/functional/rpc_net.py
+++ b/test/functional/rpc_net.py
@@ -46,11 +46,13 @@ class NetTest(BitcoinTestFramework):
         self.num_nodes = 2
         self.extra_args = [
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-minrelaytxfee=10",
             ],
             [
+                "-avalanche=1",
                 "-avaproofstakeutxodustthreshold=1000000",
                 "-avaproofstakeutxoconfirmations=1",
                 "-minrelaytxfee=5",
diff --git a/test/functional/test_framework/test_framework.py b/test/functional/test_framework/test_framework.py
index 7681219dae105f44aa08d06aab1d2b2737d6d552..1f2521c62fff59087128bd30b53049d37751d8a5 100644
--- a/test/functional/test_framework/test_framework.py
+++ b/test/functional/test_framework/test_framework.py
@@ -35,6 +35,7 @@ from .util import (
     initialize_datadir,
     p2p_port,
     rpc_port,
+    try_rpc,
     uint256_hex,
     wait_until_helper_internal,
 )
@@ -890,7 +891,18 @@ class BitcoinTestFramework(metaclass=BitcoinTestMetaClass):
     def sync_all(self, nodes=None):
         self.sync_blocks(nodes)
         self.sync_mempools(nodes)
-        self.sync_proofs(nodes)
+        # Temporary hack to check if avalanche is enabled. This assumes all nodes
+        # have avalanche either enabled or disabled.
+        if (
+            nodes
+            and len(nodes) > 0
+            and not try_rpc(
+                -32603,
+                "Avalanche processor missing or disabled",
+                nodes[0].getavalancheinfo,
+            )
+        ):
+            self.sync_proofs(nodes)
 
     def wait_until(self, test_function, timeout=60, check_interval=0.05):
         return wait_until_helper_internal(
